src/group/errors.rs

```diff
@@ -4,102 +4,107 @@
 //! `CreateCommitError`.
 
 use crate::config::ConfigError;
-use crate::extensions::ExtensionError;
 use crate::framing::errors::MLSCiphertextError;
-use crate::tree::TreeError;
+use crate::tree::{secret_tree::SecretTypeError, TreeError};
 
-#[derive(PartialEq, Debug)]
-#[repr(u16)]
-pub enum GroupError {
-    DecryptionError = 0,
-}
-
-#[derive(PartialEq, Debug)]
-#[repr(u16)]
-pub enum WelcomeError {
-    CiphersuiteMismatch = 100,
-    JoinerSecretNotFound = 101,
-    MissingRatchetTree = 102,
-    TreeHashMismatch = 103,
-    JoinerNotInTree = 104,
-    ConfirmationTagMismatch = 105,
-    InvalidRatchetTree = 106,
-    InvalidGroupInfoSignature = 107,
-    GroupInfoDecryptionFailure = 108,
-    DuplicateRatchetTreeExtension = 109,
-    UnsupportedMlsVersion = 110,
-    UnknownError = 111,
-}
-
-#[derive(PartialEq, Debug)]
-#[repr(u16)]
-pub enum ApplyCommitError {
-    EpochMismatch = 200,
-    WrongPlaintextContentType = 201,
-    SelfRemoved = 202,
-    PathKeyPackageVerificationFailure = 203,
-    NoParentHashExtension = 204,
-    ParentHashMismatch = 205,
-    PlaintextSignatureFailure = 206,
-    RequiredPathNotFound = 207,
-    ConfirmationTagMismatch = 208,
-    MissingOwnKeyPackage = 209,
-    MissingProposal = 210,
-    OwnKeyNotFound = 211,
-}
-
-#[derive(PartialEq, Debug)]
-#[repr(u16)]
-pub enum CreateCommitError {
-    CannotRemoveSelf = 300,
-    OwnKeyNotFound = 301,
-}
-
-#[derive(PartialEq, Debug)]
-#[repr(u16)]
-pub enum ExporterError {
-    KeyLengthTooLong = 400,
+implement_error! {
+    pub enum GroupError {
+        MLSCiphertextError(MLSCiphertextError) =
+            "See [`MLSCiphertextError`](`crate::framing::errors::MLSCiphertextError`) for details",
+        WelcomeError(WelcomeError) =
+            "See [`WelcomeError`](`WelcomeError`) for details",
+        ApplyCommitError(ApplyCommitError) =
+            "See [`WelcomeError`](`ApplyCommitError`) for details",
```

Comment:
Is the discrepancy in the link on purpose?

Reply:
c&p ftw